main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func main() {
+	fieldsParameter := flag.String("f", "", "Choose field to print")
+	flag.Parse()
+
+	input := strings.Split(*fieldsParameter, ",")
+	inputAsInt := make([][]int, len(input))
+
+	for i := range input {
+		split := strings.Split(input[i], "-")
+		inputAsInt[i] = make([]int, len(split))
+
+		for sid, s := range split {
+			value, err := strconv.Atoi(s)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+			inputAsInt[i][sid] = value
+		}
+
+		if len(inputAsInt[i]) > 1 && inputAsInt[i][1]-inputAsInt[i][0] > 1 {
+			for p := inputAsInt[i][0] + 1; p < inputAsInt[i][1]; p++ {
+				inputAsInt[i] = append(inputAsInt[i], p)
+			}
+			sort.Ints(inputAsInt[i])
+		}
+	}
+
+	content, readAllErr := ioutil.ReadAll(os.Stdin)
```

Comment:
You're reading all of the data into memory. What happens if you're running this on a VM w/ 1G RAM, but your in.txt is 2G large?

Reply:
then the readAllErr will be != nil  and the program exits

Reply:
No, either the Go runtime panics or the process is killed by the OS. Anyway, the standard cut wouldn't behave like this.

How would you fix this?